.vib/supabase/cypress/cypress/integration/supabase_spec.js

```diff
@@ -0,0 +1,143 @@
+/// <reference types="cypress" />
+import {
+  random
+} from '../support/utils';
+
+/*
+  Out of the possible components, these are the ones that can be tested without
+  having to create a full Supabase development project. Here we check
+
+     - The Authorization service: checking that we can create a user
+     - The Postgres Meta service: checking that we can interact with the database
+     - The Storage service: checking that we can create a bucket
+*/
+
+it('can create a user and generate a token to it (auth)', () => {
+  // Source: https://github.com/netlify/gotrue
+  cy.fixture('users').then((users) => {
+    const userPayload = {
+      email: `${random}${users.newUser.email}`,
+      password: `${users.newUser.password}-${random}`,
+    };
+    cy.request({
+      method: 'POST',
+      headers: { apikey: Cypress.env('serviceKey') },
+      url: '/auth/v1/signup',
+      body: userPayload,
+    }).then((response) => {
+      expect(response.status).to.eq(200);
+      expect(JSON.stringify(response.body)).to.have.string(userPayload.email);
+    });
+
+    cy.request({
+      method: 'POST',
+      headers: { apikey: Cypress.env('serviceKey') },
+      // This one works with URL parameters, not with a body
+      url: `/auth/v1/token?grant_type=password&username=${userPayload.email}&password=${userPayload.password}`,
+    }).then((response) => {
+      expect(response.status).to.eq(200);
+      expect(JSON.stringify(response.body)).to.have.string('access_token');
+    });
+  });
+});
+
+it('can create a schema and it appears on the list of schemas (meta)', () => {
+  // Source: https://supabase.github.io/postgres-meta/
+  cy.fixture('schemas').then((schemas) => {
+    const schemaPayload = {
+      name: `${schemas.newSchema.name}${random}`,
+      owner: schemas.newSchema.owner,
+    };
+    cy.request({
+      method: 'POST',
+      headers: { apikey: Cypress.env('serviceKey') },
+      url: '/pg/schemas',
+      body: schemaPayload,
+    }).then((response) => {
+      expect(response.status).to.eq(200);
+      expect(JSON.stringify(response.body)).to.have.string(schemaPayload.name);
+    });
+
+    cy.request({
+      method: 'GET',
+      headers: { apikey: Cypress.env('serviceKey') },
+      url: '/pg/schemas',
+    }).then((response) => {
+      expect(response.status).to.eq(200);
+      expect(JSON.stringify(response.body)).to.have.string(schemaPayload.name);
+    });
+  });
+});
+
+it('can create a bucket and upload a file (storage)', () => {
```

Comment:
This would create a test dependency because the bucket is needed for uploading the file, and they want to avoid that. @joancafom can confirm but I believe that currently it is the way they want the tests to be.

Reply:
Indeed, maintaining the tests independent from each other assures you there won't be cascade failing. Moreover, I think debugging this test is fairly easy given that you are creating a bucket, checking it, and then creating (and checking) a file.